File: 02.lineales/sort/insert-sort.go
package main

import (
	"fmt"
)

func main() {
	var numeros = []int{15, 14, 13, 12, 10, 9, 8, 7, 6, 5}
	for i := 1; i < len(numeros); i++ {
		j := i
		for j > 0 && numeros[j] < numeros[j-1] {
			fmt.Println("swap")
			tmp := numeros[j]
			numeros[j] = numeros[j-1]
			numeros[j-1] = tmp
			j = j - 1
		}
	}

	for _, m := range numeros {
		fmt.Printf("%d\t", m)
	}

}

File: 02.lineales/sort/insert-sort.c
#include<stdio.h>

#define CANTIDAD 10

int main() {
  int numeros[CANTIDAD] = {15, 14, 13, 12, 10, 9, 8, 7, 6, 5};
  int tmp;
  for (int i=1; i<CANTIDAD; i++) {
    for (int j=i;  numeros[j] < numeros[j-1]; j--) {
        
            printf("Swap!\n");
            tmp = numeros[j];
            numeros[j] = numeros[j-1];
            numeros[j-1] = tmp;
        
    }
  }

  for (int i=0; i<CANTIDAD; i++) {
    printf("%d%s", numeros[i], (i==CANTIDAD-1?"":"\t" ));
  }
  printf("\n");
  return(0);
}
